SpockOpaleye/src/ApiBase.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE FlexibleContexts       #-}
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE OverloadedStrings      #-}
+{-# LANGUAGE TemplateHaskell        #-}
+
+module ApiBase where
+
+import           Control.Lens
+import qualified Data.Profunctor.Product.Default as D
+import           Data.Time                       (UTCTime, getCurrentTime)
+import           Database.PostgreSQL.Simple
+import           DataTypes
+import           Opaleye
+import           OpaleyeDef
+import           OpaleyeTypes
+import           Prelude                         hiding (id)
+
+create_row ::(
+    HasCreatedat columnsW (Maybe (Column PGTimestamptz)),
+    HasUpdatedat columnsW (Column PGTimestamptz),
+    D.Default Constant incoming columnsW, D.Default QueryRunner returned row)
+    => Connection -> Table columnsW returned -> incoming -> IO row
+create_row conn table item = do
+  current_time <- fmap pgUTCTime getCurrentTime
+  let itemPg = (constant item) & createdat .~ (Just current_time) & updatedat .~ (current_time)
+  fmap head $ runInsertManyReturning conn table [itemPg] (\x -> x)
+
+update_row :: (
+    HasUpdatedat haskells UTCTime
+    , D.Default Constant haskells columnsW
+    , D.Default Constant item_id (Column PGInt4)
+    , HasId haskells item_id
+    , HasId columnsR (Column PGInt4)
+    )
+    => Connection -> Table columnsW columnsR -> item_id -> haskells -> IO haskells
+update_row conn table it_id item = do
```

Comment:
Passing the id separately explicitly specify which record to update. The record contains the data that the row should be updated with. The id specify the row that should be updated. Since those are separate things, I think those must remain separate.

Imagine a case when someone have a record where they have made some updates to the fields, and want to save to a certain Id. But imagine that they missed to change the id field. When they make the call, the wrong record is going to get updated. There is nothing warning the user that they didn't specify the id of the row that should be updated, because that was implicitly assumed to be the id that came with the updated record.